src/testdrive/src/parser.rs

```diff
@@ -141,6 +130,21 @@ fn parse_builtin(line_reader: &mut LineReader) -> Result<BuiltinCommand, InputEr
     })
 }
 
+/// Validate that the string is an allowed variable name (lowercase letters, numbers and dashes)
+pub fn validate_var_name(name: &str) -> Result<(), String> {
```

Comment:
`validate_var_name` seems a little too specific given that this validates both command keys and variable names. How about `validate_ident`?